internal/services/domainservices/active_directory_domain_service_trust_resource_test.go

```diff
@@ -0,0 +1,185 @@
+package domainservices_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/domainservices/mgmt/2020-01-01/aad"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/domainservices/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type DomainServiceTrustResource struct {
+	// The name of the resource group contains the AADDS
+	AADDSResourceGroupName string
+	// The name the AADDS
+	AADDSName string
+	// The FQDN of the on-premise ADDS
+	ADDSFqdn string
+	// The primary IP of the DNS that can resolve the on-premise ADDS
+	ADDSIp1 string
+	// The secondary IP of the DNS that can resolve the on-premise ADDS
+	ADDSIp2 string
+	// The password of the inbound trust set in the on-premise ADDS
+	TrustPassword string
+}
+
+func NewDomainServiceTrustResource() (*DomainServiceTrustResource, error) {
+	aaddsRgName := os.Getenv("ARM_TEST_AADDS_RESOURCE_GROUP_NAME")
+	if aaddsRgName == "" {
+		return nil, fmt.Errorf("`ARM_TEST_AADDS_RESOURCE_GROUP_NAME` is not set")
+	}
+
+	aaddsName := os.Getenv("ARM_TEST_AADDS_NAME")
+	if aaddsName == "" {
+		return nil, fmt.Errorf("`ARM_TEST_AADDS_NAME` is not set")
+	}
+
+	addsFqdn := os.Getenv("ARM_TEST_ADDS_FQDN")
+	if addsFqdn == "" {
+		return nil, fmt.Errorf("`ARM_TEST_ADDS_FQDN` is not set")
+	}
+
+	addsIp1 := os.Getenv("ARM_TEST_ADDS_IP1")
+	if addsIp1 == "" {
+		return nil, fmt.Errorf("`ARM_TEST_ADDS_IP1` is not set")
+	}
+
+	addsIp2 := os.Getenv("ARM_TEST_ADDS_IP2")
+	if addsIp2 == "" {
+		return nil, fmt.Errorf("`ARM_TEST_ADDS_IP2` is not set")
+	}
+
+	password := os.Getenv("ARM_TEST_PASSWORD")
+	if password == "" {
+		return nil, fmt.Errorf("`ARM_TEST_PASSWORD` is not set")
+	}
+
+	return &DomainServiceTrustResource{
+		AADDSResourceGroupName: aaddsRgName,
+		AADDSName:              aaddsName,
+		ADDSFqdn:               addsFqdn,
+		ADDSIp1:                addsIp1,
+		ADDSIp2:                addsIp2,
+		TrustPassword:          password,
+	}, nil
+}
+
+func TestAccDomainServiceTrust_basic(t *testing.T) {
```

Comment:
In terms of concurrency, these are probably OK for now in their current form, but we do need to look at handling concurrency with AADDS in general since it's tested in a few places now.

Reply:
What does the `a few places` mean? If they all belong to the acc test cases, then we can make all of them as sequential tests?

Reply:
There are also tests in different service packages that depend on AADDS

Reply:
Then should we also guarantee that they are running in sequential?

Reply:
That's not possible at the moment, as they are in different service packages and built/tested separately.

Reply:
I see.